src/include/stir/FilePath.h

```diff
@@ -0,0 +1,170 @@
+/*
+    Copyright (C) 2018, University of Hull
+
+    This file is part of STIR.
+    This file is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.
+
+    This file is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    See STIR/LICENSE.txt for details
+*/
+
+#ifndef __stir_FILEPATH_H__
+#define __stir_FILEPATH_H__
+
+/*!
+  \file
+  \ingroup buildblock
+
+  \brief Declaration of class stir::FilePath
+  This is a class implementing basic filesytem functionality. Parts of this class were
+  copied from Functions for filename manipulations.
+
+  Maximum effort was put in order to be cross platform.
+
+  \author Nikos Efthimiou
+
+*/
+
+#include "stir/common.h"
+#include <stdio.h>
+#include <iostream>
+#include <string>
+#include <boost/format.hpp>
+// N.E: Not sure I want this here. Included for the max_filename_length.
+#include "stir/utilities.h"
+
+#if defined(__OS_WIN__)
+	#include <windows.h>
+	 #include<sys/types.h> // required for stat.h
+	#include<direct.h>
+#else
+	#include <unistd.h>
+#endif
+
+#include <sys/stat.h>
+START_NAMESPACE_STIR
+
+/*!
+ * \ingroup buildblock
+ *
+ * \brief The FilePath class
+ *
+ * \par Description
+ * This class provides basic filesystem support for STIR. It is build in a similar manner to the functions in the utilities but extends with support to create folders, check if folder exists and is writable.
+ *
+ * The old pathname operations are supported.
+ */
+
+class FilePath
+{
+public:
+    FilePath();
+
+    //!
+    //! \brief FilePath
+    //! \param __str Name of path
+    //! \param _check Perform checks to the __str
+    //!
+    //! If the initial path exists in the disk _check should be left true.
+    //! It will check that the path is writtable
```

Comment:
"writable"

Document what will happen if the check fails. (presumably call `error()`)

If you can be bothered, you could write `\p _check` etc wherever you use a parameter (not after the `\param` keyword)

Reply:
ok